### PR TITLE
for node_map changing map<*,*> to list<pair<*,*>>

### PR DESCRIPTION
**Pull request by ysyrov**
It's pointless to sort by pointer value, plus it shuffles files contents (after read-write).
Please pull this into master.

### CODE REVIEW

File: src/node_data.cpp [hunk 1]
```diff
@@ -256,7 +256,7 @@ void node_data::reset_map() {
 }
 
 void node_data::insert_map_pair(node& key, node& value) {
-  m_map[&key] = &value;
+  m_map.push_back({ &key, &value });
```
Comment (ysyrov) [inline src/node_data.cpp]: agree, or, to be consistent with line below it:
push_back(kv_pair(&key, &value));

Comment (jbeder) [inline src/node_data.cpp]: Please go with emplace_back, and update the nearby line to be consistent.

**Comment by jbeder** [comment]
Please change the commit message to be in imperative present tense (e.g., "Change ...")

**Comment by ysyrov** [comment]
I disagree with commit message to be in imperative.
It's a description to an artifact and should reflect what it is, not what someone else needs to do.
;)

File: include/yaml-cpp/node/detail/node_data.h [hunk 2]
```diff
@@ -114,7 +114,7 @@ class YAML_CPP_API node_data {
   mutable std::size_t m_seqSize;
 
   // map
-  typedef std::map<node*, node*> node_map;
+  typedef std::list< std::pair<node*, node*> > node_map;
```
Comment (jbeder) [inline include/yaml-cpp/node/detail/node_data.h]: Make sure you run this change through clang-format. You shouldn't need the extra spaces around the < and >

**Comment by jbeder** [comment]
Can you squash these commits, and also capitalize the first letter and put a period at the end of the sentence? Also, please put a little explanation of *why* you're changing this in the commit message.

Btw, check out http://git.kernel.org/cgit/git/git.git/tree/Documentation/SubmittingPatches?id=HEAD for some discussion of the tense, e.g.

> Describe your changes in imperative mood, e.g. "make xyzzy do frotz"
> instead of "[This patch] makes xyzzy do frotz" or "[I] changed xyzzy
> to do frotz", as if you are giving orders to the codebase to change
> its behaviour.

**Comment by jbeder** [comment]
Also, make sure you rebase to HEAD, of course.

**Comment by ysyrov** [comment]
Ok, it makes sense when change is not an item, but action ("patch" can be both, BTW). Then "change" is imperative verb. Arguing semantics, aren't we?
How about:

Change node_map type from map<*,*> to list<pair<*,*>>.
This will prevent unpredictable reordering of documents after load and save.

**Comment by ysyrov** [comment]
the cb26bbf looks good

**Comment by jbeder** [comment]
You've got 16 commits in this pull request. Just put one in and rebase against HEAD.

File: include/yaml-cpp/node/detail/node_data.h [hunk 3]
```diff
@@ -114,7 +114,7 @@ class YAML_CPP_API node_data {
   mutable std::size_t m_seqSize;
 
   // map
-  typedef std::map<node*, node*> node_map;
+  typedef std::list<std::pair<node*, node*> > node_map;
```
Comment (jbeder) [inline include/yaml-cpp/node/detail/node_data.h]: Extra space between > >

File: include/yaml-cpp/node/detail/node_iterator.h [hunk 4]
```diff
@@ -37,7 +37,7 @@ struct node_iterator_value : public std::pair<V*, V*> {
 };
 
 typedef std::vector<node*> node_seq;
-typedef std::map<node*, node*> node_map;
+typedef std::list<std::pair<node*, node*> > node_map;
```
Comment (jbeder) [inline include/yaml-cpp/node/detail/node_iterator.h]: ditto

File: src/node_data.cpp [hunk 5]
```diff
@@ -256,9 +256,9 @@ void node_data::reset_map() {
 }
 
 void node_data::insert_map_pair(node& key, node& value) {
-  m_map[&key] = &value;
+  m_map.emplace_back(kv_pair(&key, &value));
```
Comment (jbeder) [inline src/node_data.cpp]: You can just emplace_back(&key, &value), right?

File: include/yaml-cpp/node/detail/node_data.h (hunk 3 above)
Comment (ysyrov) [inline include/yaml-cpp/node/detail/node_data.h]: that's clang-format

File: src/node_data.cpp (hunk 5 above)
Comment (ysyrov) [inline src/node_data.cpp]: yes, all compiles to same thing.
I personally prefer {}

File: include/yaml-cpp/node/detail/node_data.h (hunk 3 above)
Comment (jbeder) [inline include/yaml-cpp/node/detail/node_data.h]: My bad, I didn't make sure the style used C++ 11. Please try again at HEAD, it should adjust it to >>

File: src/node_data.cpp (hunk 5 above)
Comment (jbeder) [inline src/node_data.cpp]: I don't understand what you're saying

```
emplace_back(kv_pair(&key, &value))
```

and

```
emplace_back(&key, &value)
```

don't compile to the same thing. The former uses pair's copy-constructor and the latter uses pair's two-arg constructor.

**Pull request closed without merging**